File: 0290-word-pattern/0290-word-pattern.js
/**
 * @param {string} pattern
 * @param {string} s
 * @return {boolean}
 */
var wordPattern = function(pattern, s) {
    s = s.split(" ");
    if (new Set(pattern.split("")).size !== new Set(s).size) return false;
    let map = {};
    for (let i=0; i<s.length; i++)
    {
        if (pattern.charAt(i) in map && map[pattern.charAt(i)] !== s[i]) return false;
        map[pattern.charAt(i)] = s[i];
    } 
    const values = Object.values(map);
    return new Set(s).size === Object.values(map).length;
};

File: 0290-word-pattern/0290-word-pattern.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        s = s.split(" ")
        if len(pattern) != len(s): return False
        mapPS = {}
        mapSP = {}

        for i in range(0, len(s)):
            if pattern[i] in mapPS and mapPS[pattern[i]] != s[i]:
                return False
            else: mapPS[pattern[i]] = s[i]
            
            if s[i] in mapSP and mapSP[s[i]] != pattern[i]:
                return False
            else: mapSP[s[i]] = pattern[i]

        return True